scripts/core/editor3/reducers/suggestions.jsx

```diff
@@ -206,26 +247,41 @@ const createChangeBlockStyleSuggestion = (state, {blockType, data}) => {
  */
 const pasteAddSuggestion = (state, {content, data}) => {
     let {editorState} = state;
-    const blockMap = content.getBlockMap();
+    const initialSelection = editorState.getSelection();
 
-    editorState = deleteCurrentSelection(editorState, data);
+    const beforeStyle = Highlights.getHighlightStyleAtOffset(editorState, changeSuggestionsTypes, initialSelection, -1);
+    const beforeData = beforeStyle != null ? Highlights.getHighlightData(editorState, beforeStyle) : null;
 
-    blockMap.forEach((block) => {
-        if (block.getType() !== 'atomic') {
-            const text = block.getText();
-            const characterMetadataList = block.getCharacterList();
-            let inlineStyle;
+    // add content to editor state
+    const mergedContent = Modifier.replaceWithFragment(
+        editorState.getCurrentContent(),
+        editorState.getSelection(),
+        sanitizeContent(content).getBlockMap()
+    );
 
-            characterMetadataList.forEach((characterMetadata, i) => {
-                inlineStyle = acceptedInlineStyles.filter((style) => characterMetadata.hasStyle(style));
-                editorState = setAddSuggestionForCharacter(editorState, data, text[i], inlineStyle);
-            });
-        } else {
-            // TODO: insert block
-        }
-    });
+    // push new content
+    editorState = EditorState.push(editorState, mergedContent, 'insert-fragment');
 
-    return saveEditorStatus(state, editorState, 'change-block-data');
+    // store current selection for later
+    const finalSelection = editorState.getSelection();
+
+    // select pasted content
+    editorState = EditorState.acceptSelection(editorState, editorState.getSelection().merge({
+        anchorKey: initialSelection.getAnchorKey(),
```

Comment:
Is this working well when the initial selection is backward?
Usually I use getStartKey/Offset and getEndKey/Offset and isBackward: false.

Reply:
I think it doesn't, will push an update